In [7]:
!pip install pandas

  Using cached pytz-2020.1-py2.py3-none-any.whl (510 kB)


In [14]:
!pwd

/c/Users/�씠�삙�꽦/Desktop/data_anal/TDA


In [15]:
!ls

answer
ch01.ipynb


In [8]:
%matplotlib inline

In [16]:
import pandas as pd
import numpy as np
import matplotlib.pyplot as plt 

file_path = 'answer/data/drinks.csv'
drinks = pd.read_csv(file_path) #DataFrame형태로 호출

In [17]:
print(drinks.info()) # 대략적으로 어떤 정보를 가지고 있는지 확인


<class 'pandas.core.frame.DataFrame'>
RangeIndex: 193 entries, 0 to 192
Data columns (total 6 columns):
 #   Column                        Non-Null Count  Dtype  
---  ------                        --------------  -----  
 0   country                       193 non-null    object 
 1   beer_servings                 193 non-null    int64  
 2   spirit_servings               193 non-null    int64  
 3   wine_servings                 193 non-null    int64  
 4   total_litres_of_pure_alcohol  193 non-null    float64
 5   continent                     170 non-null    object 
dtypes: float64(1), int64(3), object(2)
memory usage: 9.2+ KB
None


위 데이터 간단히 살펴보기 
193개의 행 6개의 열로 이루어진 데이터 프레임입니다. 

그중 0번,5번 인덱스를 제외하고는 정수와 실수로 이루어진 인덱스(1~4)입니다. 

주의할 것이 5번인덱스(continent)는 23개의 데이터가 비어있습니다.


In [20]:
# 데이터 모습을 확인해볼게요.
drinks.head()

country  beer_servings  spirit_servings  wine_servings  \
0  Afghanistan              0                0              0   
1      Albania             89              132             54   
2      Algeria             25                0             14   
3      Andorra            245              138            312   
4       Angola            217               57             45   

   total_litres_of_pure_alcohol continent  
0                           0.0        AS  
1                           4.9        EU  
2                           0.7        AF  
3                          12.4        EU  
4                           5.9        AF

In [21]:
# 숫자정보(정수, 실수)의 통계정보(평균, 표준편차, 분산등) 확인
drinks.describe()

beer_servings  spirit_servings  wine_servings  \
count     193.000000       193.000000     193.000000   
mean      106.160622        80.994819      49.450777   
std       101.143103        88.284312      79.697598   
min         0.000000         0.000000       0.000000   
25%        20.000000         4.000000       1.000000   
50%        76.000000        56.000000       8.000000   
75%       188.000000       128.000000      59.000000   
max       376.000000       438.000000     370.000000   

       total_litres_of_pure_alcohol  
count                    193.000000  
mean                       4.717098  
std                        3.773298  
min                        0.000000  
25%                        1.300000  
50%                        4.200000  
75%                        7.200000  
max                       14.400000

In [22]:
# 'beer_serving', 'wine_servings' 두 피처간의 상관계수를 계산합니다. 
# pearson은 상관계수를 구하는 방법 중 하나를 의미하며, 가장 널리 쓰이는 방법이에요. 
corr = drinks[['beer_servings','wine_servings']].corr(method='pearson')
corr

# 1에 가깔울수록 서로 관계가 깊고 반대로 0에 가까울수록 관계가 낮아진다.

beer_servings  wine_servings
beer_servings       1.000000       0.527172
wine_servings       0.527172       1.000000

---

[여러 피처의 상관관계 분석]

In [24]:
# 피처간의 상관계수 행렬을 구합니다. 
cols = ['beer_servings', 'spirit_servings', 'wine_servings', 'total_litres_of_pure_alcohol']
corr = drinks[cols].corr(method = 'pearson')
print(corr)

                              beer_servings  spirit_servings  wine_servings  \
beer_servings                      1.000000         0.458819       0.527172   
spirit_servings                    0.458819         1.000000       0.194797   
wine_servings                      0.527172         0.194797       1.000000   
total_litres_of_pure_alcohol       0.835839         0.654968       0.667598   

                              total_litres_of_pure_alcohol  
beer_servings                                     0.835839  
spirit_servings                                   0.654968  
wine_servings                                     0.667598  
total_litres_of_pure_alcohol                      1.000000  
